### ISSUE TITLE
Feature to bulk whitelist url in settings.

### ISSUE THREAD

**Issue by vipinkant-winzo** [httptoolkit/httptoolkit#545]
Feature to bulk whitelist url in settings is missing. We need to whitelist each url individually.

---

_Does this affect you too? Click below and add a :+1: to vote for this and help decide where HTTP Toolkit goes next, or [go vote on the other most popular ideas so far](https://github.com/httptoolkit/feedback/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc)._

**Comment by pimterry** [comment]
Interesting idea, I haven't heard this suggested before.

Can you explain more about why you have so many URLs you want to whitelist?

What format would you want to provide the whitelist in?

**Comment by lord-vipin** [comment]
For our IOS app specifically, In some cases We have some SDK's in the app or some third party api which are getting called from app. So I don't necessarily want to inspect them and due to security reasons these calls got giving error if not whitelisted.

**Comment by pimterry** [comment]
Ah, so your third-party APIs talk to servers that have invalid HTTPS configurations? Or they use self-signed certificates?

It depends on your setup, but if this is because they're all using a separate CA (an internal one or similar) you can skip the whitelist entirely, and just export the CA certificate and then add that to the "Trusted CA Certificates" section in the settings. Would that work for you?

Are all these URLs related, so they'd match some kind of pattern, like `*.example.com`? I've been thinking about adding wildcard support here. In the extreme case, you could just add `*` to the whitelist, but of course that's not very secure (it would effectively disable HTTPS validation for all requests).

I'll look into CSV support separately anyway, it'd take a little time but that should be possible in future.

**Comment by lord-vipin** [comment]
Yeah wildcard support for whitelisting domain would also be a great improvement.

**Comment by lord-vipin** [comment]
I do not have the option to export the CA certificate, of our domain to add it to the trusted CA certificate in settings, that is why I am using this option of adding the domains of the third-party libs to bypass them, third party libraries example are firebase, clevertap, etc they use their trusted CA certificate.

**Comment by pimterry** [comment]
> I do not have the option to export the CA certificate

Ok, so there is just one or a small number of CA certificates involved here?

In that case, trusting the certificate will be much easier and you don't need special tools to do this.

You can get the certificate by opening the URL in a browser, for example, viewing the certificate there via the URL bar, selecting the CA and exporting.

Alternatively, you can do this with OpenSSL directly:

```
openssl s_client -showcerts -servername www.example.com -connect www.example.com:443 </dev/null
```

That will print the certificate chain for the site (e.g. www.example.com). You just need to save one certificate chunk (the lines from BEGIN CERTIFICATE to END CERTIFICATE) into a .pem file. In the output, the first certificate will be the site (CN = www.example.com), and the later certificate(s) will be the CAs.

Once you have a PEM file containing the CA certificate, if you add that as a trusted CA in HTTP Toolkit's settings everything will work for all domains immediately. This is also more secure, because HTTPS whitelisting disables many security checks, while trusting different CAs does not.
